codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostBodyUtil.java

```diff
@@ -172,6 +172,34 @@ static int findLineBreak(ByteBuf buffer, int index) {
         return posFirstChar;
     }
 
+    /**
+     * Try to find last LF or CRLF as Line Breaking
+     *
+     * @param buffer the buffer to search in
+     * @param index the index to start from in the buffer
+     * @return a relative position from index > 0 if LF or CRLF is found
+     *         or < 0 if not found
+     */
+    static int findLastLineBreak(ByteBuf buffer, int index) {
+        int toRead = buffer.readableBytes() - (index - buffer.readerIndex());
+        int posFirstChar = buffer.bytesBefore(index, toRead, HttpConstants.LF);
```

Comment:
Since we scan start to end, this method can be implemented in terms of `findLineBreak`, right?

Reply:
Let me see (double check) if those two methods are indeed the same, just by changing the index value.

I'm not that sure since `fineLineBreak` finds the "first" one, while `findLastLineBreak` finds the "last" one (so the recursion for the moment). And as it is not optimized to make a reverse reading (we cannot use `bytesBefore`)...

Reply:
Not the same (first vs last)

Reply:
I was thinking something like this:

```java
    static int findLastLineBreak(ByteBuf buffer, int index) {
        int candidate = findLineBreak(buffer, index);
        int next;
        while (candidate >= 0 && (next = findLineBreak(buffer, index + candidate)) > 0) {
            candidate += next;
        }
        return candidate;
    }
```

Any reason that can't work?

Reply:
Yes indeed, but almost since we need to recheck if there were CR at first pos, or LF, in order to restart at candudate + 2 or +1.
However simplest than current version, I agree.

Reply:
I will change and check of course.

Reply:
@chrisvest  I checked with this version working. Is it ok for you ?

    static int findLastLineBreak(ByteBuf buffer, int index) {
        int candidate = findLineBreak(buffer, index);
        int findCRLF = 0;
        if (candidate >= 0) {
            if (buffer.getByte(index + candidate) == HttpConstants.CR) {
                findCRLF = 2;
            } else {
                findCRLF = 1;
            }
            candidate += findCRLF;
        }
        int next;
        while (candidate > 0 && (next = findLineBreak(buffer, index + candidate)) >= 0) {
            candidate += next;
            if (buffer.getByte(index + candidate) == HttpConstants.CR) {
                findCRLF = 2;
            } else {
                findCRLF = 1;
            }
            candidate += findCRLF;
        }
        return candidate - findCRLF;
    }

Reply:
Yes, that looks good. Thanks for explaining.

Reply:
Commited ;-)